internal/core/server/workspace.go

```diff
@@ -747,6 +760,84 @@ func (ws *Workspace) DescribeNetworks(ctx context.Context, input *api.DescribeNe
 	return &output, nil
 }
 
+func (ws *Workspace) ExportProcfile(ctx context.Context, input *api.ExportProcfileInput) (*api.ExportProcfileOutput, error) {
+	procs, err := ws.DescribeProcesses(ctx, &api.DescribeProcessesInput{})
+	if err != nil {
+		return nil, fmt.Errorf("describing processes: %w", err)
+	}
+
+	unixProcs := make([]procfile.Process, 0, len(procs.Processes))
+	for _, proc := range procs.Processes {
+		if proc.Provider == "unix" {
+			spec := proc.Spec.(process.Spec)
+
+			unixProcs = append(unixProcs, procfile.Process{
+				Name:        proc.Name,
+				Program:     spec.Program,
+				Arguments:   spec.Arguments,
+				Environment: spec.Environment,
+			})
+		}
+	}
+
+	export, err := procfile.Generate(unixProcs)
+	if err != nil {
+		return nil, fmt.Errorf("generating procfile: %w", err)
+	}
+
+	return &api.ExportProcfileOutput{
+		Procfile: export,
+	}, nil
+}
+
+func (ws *Workspace) ReadFile(ctx context.Context, input *api.ReadFileInput) (*api.ReadFileOutput, error) {
+	resolvedPath, err := ws.resolveWorkspacePath(ctx, input.Path)
+	if err != nil {
+		return nil, err
+	}
+
+	content, err := os.ReadFile(resolvedPath)
+	if err != nil {
+		return nil, fmt.Errorf("reading file: %w", err)
```

Comment:
probably don't want the `Errorf` here, as this is really the main crux of the `ReadFile` function & so doesn't add real context. Callers might wind up with `reading file: reading file: ...`